lib/plugins/aws/remove/lib/bucket.js

```diff
@@ -16,17 +16,29 @@ module.exports = {
     const serviceStage = `${this.serverless.service.service}/${this.provider.getStage()}`;
 
     return this.provider
-      .request('S3', 'listObjectsV2', {
+      .request('S3', 'listObjectVersions', {
```

Comment:
I have concerns when it comes to universally using this different method. It might potentially cause unexpected issues for people that are using granular IAM permissions and do not grant access to this method (and grant to `listObjectsV2`), which might make it a breaking change of sort. What do you think @medikoo? Should we treat such situations as breaking changes or not as it only concerns potential issues with permissions?

Reply:
> If that's the case, maybe we should use listObjectVersions only with deploymentBucketObject.versioning: true ?

+1 on that, if we can avoid a risk of BC break (for restricted IAM permissions) that's great

Reply:
That's a good point on IAM permission. Unfortunately, the only way to find all versioned objects in a S3 bucket is via `listObjectVersions`, and you can't drop a bucket without first deleting these objects explicitly. The API will simply return all objects in a non-versioned bucket so functionally the same API would work in both cases.

Using `listObjectVersions` only when `deploymentBucketObject.versioning: true` is also non-ideal as there's no guarantee this setting is sync'ed with bucket's actual setting.

Reply:
@mars-lan what happens if we use `listObjectsV2` on versioned bucket?

Also I believe in such case we discuss scenario where user externally changed the setting on a bucket (and not that versioning was applied through regular deployment)

Reply:
Using `listObjectsV2` on versioned bucket will only return the "current version" objects. After deleting those you're still left with older versions and can't drop the bucket.

You're correct. We're talking about changes outside the normal deployment flow... assuming all deployments are strictly sequential. It quickly falls apart when you have multiple developers working on the same project.

Reply:
Ok, so as discussed, to not introduce a breaking version, let's use `listObjectVersions` only when `deploymentBucketObject.versioning: true`. It should be safe as long as users do not apply configuration changes externally (and in such case anything can happen)

Reply:
Fair enough. Will make changes accrodingly.

Reply:
In this spirit, as a casual passer-by here who has super granular IAM rules, I would just like to share the idea that it would be great to document what access is needed and in which cases during deployment /and/ teardown. Thanks for all the great work!

Reply:
@medikoo any recommendation on where to document this?

Reply:
@ronkorving @mars-lan I think we do not have such granular documentation anywhere, and it'll be about copying documentation that is exposed on AWS side, so I'm not sure if we should go with it (?)

Reply:
@medikoo I guess there's a choice as to how detailed you want to be. But I don't know which API calls it will make, so I also don't know what AWS documentation to consult. It's trial and error now. Even just knowing what Actions are in play during which conditions, without any further documentation, would be quite helpful.

Reply:
Thanks @ronkorving for the explanations. Still, I think this should be a general matter addressed in the context of new issue. As so far we do not practice anywhere providing fine details on what IAM rules need to be set for given functionality and settings to work.

Can you open a new issue dedicated just for that?